dal/userData.go

```diff
@@ -0,0 +1,15 @@
+package dal
+
+type UserDater interface {
+	InitClient()
+	AddActivity(username string, req string)
+	CreateNewClientActivity(username string, req string)
+	UpdateClientActivity(username string, req string)
```

Comment:
not supposed to be part of the interface

the `AddActivity` function should handle those cases, You have defined your interface according to the logic you have chosen to implement in Redis.

An interface defines a general API rather than a specific implementation of a solution